tests/test_scripts.py

```diff
@@ -6,24 +6,50 @@
 """
 
 import os.path as osp
+import sys
 import tempfile
 from collections import Counter
 from typing import List, Optional
+from unittest import mock
 
 import pandas as pd
 import pytest
 import ray.tune as tune
+import sacred
 
-from imitation.scripts.analyze import analysis_ex
-from imitation.scripts.eval_policy import eval_policy_ex
-from imitation.scripts.expert_demos import expert_demos_ex
-from imitation.scripts.parallel import parallel_ex
-from imitation.scripts.train_adversarial import train_ex
+from imitation.scripts import (
+    analyze,
+    eval_policy,
+    expert_demos,
+    parallel,
+    train_adversarial,
+)
+
+ALL_SCRIPTS_MODS = [analyze, eval_policy, expert_demos, parallel, train_adversarial]
+
+
+@pytest.fixture(autouse=True)
+def sacred_capture_use_sys():
+    """Set Sacred capture mode to "sys" because default "fd" option leads to error.
+
+    See https://github.com/IDSIA/sacred/issues/289."""
+    temp = sacred.SETTINGS["CAPTURE_MODE"]
+    sacred.SETTINGS.CAPTURE_MODE = "sys"
+    yield
+    sacred.SETTINGS.CAPTURE_MODE = temp
+
+
+@pytest.mark.parametrize("script_mod", ALL_SCRIPTS_MODS)
+def test_main_console(script_mod):
```

Comment:
Should we remove the `# pragma: no cover` lines then from the main console blocks?

Reply:
`main_console` never had `pragma: no cover`. Did you mean the `if __name__ == "__main__"` blocks?

`test_main_console` doesn't account for that, as it calls `main_console()` directly.

Reply:
If coverage for the __main__ block is something we want to have, I can add that test in a separate PR.